components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/ConfigurationDAO.java

```diff
@@ -71,6 +71,39 @@ Resource getResourceByName(int tenantId, String resourceTypeId, String name)
      */
     Resource getResourceById(String resourceId) throws ConfigurationManagementException;
 
+    /**
+     * Returns {@link Resource} by id given the tenant id.
+     *
+     * @param tenantId   Tenant id of the {@link Resource}.
+     * @param resourceId Id value of the {@link Resource} to be returned.
+     * @return {@link Resource} for the given name.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+    default Resource getTenantResourceById(int tenantId, String resourceId) throws ConfigurationManagementException {
+
+        return null;
```

Comment:
Shall we throw `NotImplementedException` from Apache commons?

Reply:
We have implemented this method in the child class, `ConfigurationDAOImpl.java` where this interface is implemented. Therefore we don't need to have default methods at this level.